earn/src/components/borrow/modal/AddCollateralModal.tsx

```diff
@@ -0,0 +1,275 @@
+
+  const numericCollateralAmount = Number(collateralAmount) || 0;
+
+  const collateralAmountBig = useMemo(() => {
+    return new Big(numericCollateralAmount).mul(10 ** collateralToken.decimals);
+  }, [collateralToken.decimals, numericCollateralAmount]);
+
+  const existingCollateralBig = useMemo(() => {
+    return new Big(existingCollateral).mul(10 ** collateralToken.decimals);
+  }, [collateralToken.decimals, existingCollateral]);
+
+  const numericUserBalance = Number(userBalance?.formatted ?? 0) || 0;
+  const userBalanceBig = useMemo(
+    () => new Big(numericUserBalance).mul(10 ** collateralToken.decimals),
+    [collateralToken.decimals, numericUserBalance]
+  );
+
+  if (!userAddress || !isOpen) {
+    return null;
+  }
+
+  const newCollateralBig = existingCollateralBig.add(collateralAmountBig).div(10 ** collateralToken.decimals);
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      title='Add Collateral'
+      setIsOpen={(open: boolean) => {
+        setIsOpen(open);
+        if (!open) {
+          resetModal();
+        }
+      }}
+      maxHeight='650px'
+    >
+      <div className='flex flex-col items-center justify-center gap-8 w-full mt-2'>
+        <div className='flex flex-col gap-1 w-full'>
+          <div className='flex flex-row justify-between mb-1'>
+            <Text size='M' weight='bold'>
+              Collateral Amount
+            </Text>
+            <BaseMaxButton
+              size='L'
+              onClick={() => {
+                if (userBalance != null) {
+                  setCollateralAmount(userBalance?.formatted);
+                }
+              }}
+            >
+              MAX
+            </BaseMaxButton>
+          </div>
+          <TokenAmountSelectInput
+            inputValue={collateralAmount}
+            onChange={(value) => {
+              const output = formatNumberInput(value);
+              if (output != null) {
+                const truncatedOutput = truncateDecimals(output, collateralToken.decimals);
+                setCollateralAmount(truncatedOutput);
+              }
+            }}
+            onSelect={(option: Token) => {
+              setCollateralAmount('');
+              setCollateralToken(option);
+            }}
+            options={tokenOptions}
+            selectedOption={collateralToken}
+          />
+        </div>
+        <div className='flex flex-col gap-1 w-full'>
+          <Text size='M' weight='bold'>
+            Summary
+          </Text>
+          <Text size='XS' color={SECONDARY_COLOR} className='overflow-hidden text-ellipsis'>
+            You're adding{' '}
+            <strong>
+              {collateralAmount || '0.00'} {collateralToken.ticker}
+            </strong>{' '}
+            as collateral to the{' '}
+            <strong>
+              {marginAccount.token0.ticker}/{marginAccount.token1.ticker}
+            </strong>{' '}
+            pair. Your total collateral for this token in this pair will be{' '}
```

Comment:
I think we should rephrase this a bit. Because they're not adding collateral to the _pair_, but rather to this particular smart wallet. For example, they may have 2 different smart wallets for the ETH/OP pair, in which case this is adding collateral to just 1 of their wallets, not both.